Automattic/pocket-casts-ios
# Categories Redesign: Modal picker

Adds a bottom sheet modal picker to select additional categories not in the pills.

https://github.com/Automattic/pocket-casts-ios/assets/3250/db647d61-b50e-41e0-b5db-efb49fdaeec6

## To test

* Build using the Staging build configuration
* Ensure the `categoriesRedesign` flag is on
* * Navigate to the Discover tab
* Tap the "All Categories" pill
* Notice the Modal which appear
* Tap a category in the Modal
* Notice that the Discover screen changes to a Category list but the Category Pills remain at the top and properly animate
* Tap the X button to reset and navigate back to the original Discover page

## Checklist
